Type error: Type 'string' not assignable to 'string[]' in opponent-deck-generator.ts

## Description

TypeScript is reporting multiple type errors in \`src/lib/opponent-deck-generator.ts\` (lines 53, 56, 57, 60, 61, 119, 125, 132, 158):

```
error TS2322: Type 'string' is not assignable to type 'string[]'.
error TS2488: Type 'Record<string, string[]>' must have a '[Symbol.iterator]()' method that returns an iterator.
```

## Acceptance Criteria

- [ ] Fix type assignments in opponent-deck-generator.ts to ensure string[] types are properly used

--- a/src/components/swiss-pairing.tsx
+++ b/src/components/swiss-pairing.tsx
@@ -613,68 +613,79 @@ export function useSwissTournament(): UseSwissTournamentReturn {
     setTournament((prev) => {
       if (!prev) return null;
 
+      // Find the match to update
+      const matchToUpdate = prev.rounds
+        .flatMap(round => round.matches)
+        .find(m => m.id === matchId);
+
+      if (!matchToUpdate) return prev;
+
+      const player1 = prev.players.find(p => p.id === matchToUpdate.player1?.id);
+      const player2 = prev.players.find(p => p.id === matchToUpdate.player2?.id);
+
+      let winner: SwissPlayer | undefined;
+      let player1Points = 0;
+      let player2Points = 0;
+
+      if (result === 'player1-win' && player1) {
+        winner = player1;
+        player1Points = 3;
+        player2Points = 0;
+      } else if (result === 'player2-win' && player2) {
+        winner = player2;
+        player1Points = 0;
+        player2Points = 3;
+      } else {
+        // Draw
+        player1Points = 1;
+        player2Points = 1;
+      }
+
+      // Update player stats
+      const updatedPlayers = prev.players.map(p => {
+        if (p.id === player1?.id) {
+          const matchResult: 'win' | 'loss' | 'draw' = 
+            result === 'player1-win' ? 'win' : result === 'player2-win' ? 'loss' : 'draw';
+          return {
+            ...p,
+            points: p.points + player1Points,
+            wins: p.wins + (result === 'player1-win' ? 1 : 0),
+            losses: p.losses + (result === 'player2-win' ? 1 : 0),
+            draws: p.draws + (result === 'draw' ? 1 : 0),
+            matchHistory: [...p.matchHistory, {
+              round: prev.currentRound,
+              result: matchResult,
+              opponentId: player2?.id || ''
+            }]
+          };
+        }
+        if (p.id === player2?.id) {
+          const matchResult: 'win' | 'loss' | 'draw' = 
+            result === 'player2-win' ? 'win' : result === 'player1-win' ? 'loss' : 'draw';
+          return {
+            ...p,
+            points: p.points + player2Points,
+            wins: p.wins + (result === 'player2-win' ? 1 : 0),
+            losses: p.losses + (result === 'player1-win' ? 1 : 0),
+            draws: p.draws + (result === 'draw' ? 1 : 0),
+            matchHistory: [...p.matchHistory, {
+              round: prev.currentRound,
+              result: matchResult,
+              opponentId: player1?.id || ''
+            }]
+          };
+        }
+        return p;
+      });
+
       const newRounds = prev.rounds.map((round) => ({
         ...round,
         matches: round.matches.map((match) => {
           if (match.id !== matchId) return match;
 
-          const player1 = prev.players.find(p => p.id === match.player1?.id);
-          const player2 = prev.players.find(p => p.id === match.player2?.id);
-          
-          let winner: SwissPlayer | undefined;
-          let player1Points = 0;
-          let player2Points = 0;
-
-          if (result === 'player1-win' && player1) {
-            winner = player1;
-            player1Points = 3;
-            player2Points = 0;
-          } else if (result === 'player2-win' && player2) {
-            winner = player2;
-            player1Points = 0;
-            player2Points = 3;
-          } else {
-            // Draw
-            player1Points = 1;
-            player2Points = 1;
-          }
-
-          // Update player stats
-          const updatedPlayers = prev.players.map(p => {
-            if (p.id === player1?.id) {
-              return {
-                ...p,
-                points: p.points + player1Points,
-                wins: p.wins + (result === 'player1-win' ? 1 : 0),
-                losses: p.losses + (result === 'player2-win' ? 1 : 0),
-                draws: p.draws + (result === 'draw' ? 1 : 0),
-                matchHistory: [...p.matchHistory, { 
-                  round: prev.currentRound, 
-                  result: result === 'player1-win' ? 'win' : result === 'player2-win' ? 'loss' : 'draw',
-                  opponentId: player2?.id || ''
-                }]
-              };
-            }
-            if (p.id === player2?.id) {
-              return {
-                ...p,
-                points: p.points + player2Points,
-                wins: p.wins + (result === 'player2-win' ? 1 : 0),
-                losses: p.losses + (result === 'player1-win' ? 1 : 0),
-                draws: p.draws + (result === 'draw' ? 1 : 0),
-                matchHistory: [...p.matchHistory, { 
-                  round: prev.currentRound, 
-                  result: result === 'player2-win' ? 'win' : result === 'player1-win' ? 'loss' : 'draw',
-                  opponentId: player1?.id || ''
-                }]
-              };
-            }
-            return p;
-          });
-
-          return { 
-            ...match, 
-            winner, 
+          return {
+            ...match,
+            winner,
             status: result === 'draw' ? 'draw' as const : 'completed' as const,
             player1Points,
             player2Points

--- a/src/components/zone-viewer.tsx
+++ b/src/components/zone-viewer.tsx
@@ -154,7 +154,7 @@ const ZoneCardList = memo(function ZoneCardList({
                 <div 
                   key={color} 
                   className={cn("w-full", colorMap[color] || "bg-gray-400")}
-                  style={{ height: `${100 / card.colors.length}%` }}
+                  style={{ height: `${100 / (card.colors?.length ?? 1)}%` }}
                 />
               ))}
             </div>

--- a/src/lib/multiplayer-layout.ts
+++ b/src/lib/multiplayer-layout.ts
@@ -109,12 +109,21 @@ export function isAdjacentPosition(
       bottom: ['top-left', 'top-right'],
       'top-left': ['bottom', 'top-right'],
       'top-right': ['bottom', 'top-left'],
+      top: [],
+      left: [],
+      right: [],
+      'bottom-left': [],
+      'bottom-right': [],
     },
     4: {
       bottom: ['left', 'right'],
       left: ['bottom', 'top'],
       top: ['left', 'right'],
       right: ['top', 'bottom'],
+      'top-left': [],
+      'top-right': [],
+      'bottom-left': [],
+      'bottom-right': [],
     },
   };
   

--- a/src/lib/opponent-deck-generator.ts
+++ b/src/lib/opponent-deck-generator.ts
@@ -139,7 +139,8 @@ function getCardsForColors(colorIdentity: string[]): { creatures: string[]; spel
 
 // Generate deck based on archetype
 export function generateOpponentDeck(input: OpponentDeckGenerationInput): GeneratedDeck {
-  const { format, archetype = 'midrange', colorIdentity = ['W', 'U', 'B', 'R', 'G'].slice(0, Math.floor(Math.random() * 5) + 1) as ['W', 'U', 'B', 'R', 'G'][], powerLevel = 'casual' } = input;
+  const defaultColors: string[] = ['W', 'U', 'B', 'R', 'G'].slice(0, Math.floor(Math.random() * 5) + 1);
+  const { format, archetype = 'midrange', colorIdentity = defaultColors, powerLevel = 'casual' } = input;
   
   const config = ARCHETYPE_CONFIGS[archetype];
   const cards: Array<{ name: string; quantity: number }> = [];
@@ -149,7 +150,8 @@ export function generateOpponentDeck(input: OpponentDeckGenerationInput): Genera
   
   // Add lands (37-40 for commander)
   const landCount = format === 'commander' ? 38 : 24;
-  const basicLands = colorIdentity.map(c => CARD_POOL.lands_basic[['W', 'U', 'B', 'R', 'G'].indexOf(c)] || 'Plains');
+  const colorIndices: Record<string, number> = { W: 0, U: 1, B: 2, R: 3, G: 4 };
+  const basicLands = colorIdentity.map(c => CARD_POOL.lands_basic[colorIndices[c]] || 'Plains');
   
   // Distribute basic lands
   for (const land of basicLands) {

